### PR TITLE
rename decimal to Decimal128

### PR DESCRIPTION
**Pull request by blagoev**

### CODE REVIEW

File: tests/js/list-tests.js [hunk 1]
```diff
@@ -202,8 +202,8 @@ module.exports = {
         TestCase.assertSimilar('string', prim.optString[0], 'a');
         TestCase.assertSimilar('data', prim.optData[0], DATA1);
         TestCase.assertSimilar('date', prim.optDate[0], new Date(1));
-        TestCase.assertSimilar('decimal', prim.optDecimal[0], Decimal128.fromString('1'));
-        TestCase.assertSimilar('object id', prim.optOid[0], new ObjectId('0000002a9a7969d24bea4cf2'));
+        TestCase.assertSimilar('decimal128', prim.optDecimal128[0], Decimal128.fromString('1'));
+        TestCase.assertSimilar('objectId', prim.optObjectId128[0], new ObjectId('0000002a9a7969d24bea4cf2'));
```
Comment (kneth) [inline tests/js/list-tests.js:206]:
```suggestion
        TestCase.assertSimilar('objectId', prim.optObjectId[0], new ObjectId('0000002a9a7969d24bea4cf2'));
```

**Comment by blagoev** [comment]
> We might have to update `CHANGELOG.md` too.

@kneth what changes are needed there

**Comment by kneth** [comment]
@blagoev `CHANGELOG.md`  says `Add support for the Decimal128 data type.`. Shouldn't it be `Add support for the decimal128 data type.`? Likewise for objectId.

**Comment by blagoev** [comment]
@kneth  I did not changed this since its `Decimal128` class name from BSON and it will get tested with `obj instanceof Decimal128`.

File: tests/js/list-tests.js [hunk 2]
```diff
@@ -1348,7 +1348,7 @@ module.exports = {
     testListNestedFromJSON: function() {
         let json = '{"id":1, "name": [{ "family": "Larsen", "given": ["Hans", "Jørgen"], "prefix": [] }, { "family": "Hansen", "given": ["Ib"], "prefix": [] }] }';
         let parent = JSON.parse(json);
-        const realm = new Realm({schema: [schemas.ParentObjectLocal, schemas.NameObjectLocal]});
+        const realm = new Realm({schema: [schemas.ParentObject, schemas.NameObject]});
```
Comment (kneth) [inline tests/js/list-tests.js:1351]: I introduced `ParentObjectLocal` and `NameObjectLocal` to make the test easier. `ParentObject` and `NameObject` are for sync and have primary keys.

Comment (blagoev) [reply tests/js/list-tests.js:1351]: let me change it quick

**Pull request merged**